soh/src/overlays/actors/ovl_En_Cow/z_en_cow.c

```diff
@@ -209,12 +226,57 @@ void func_809DF730(EnCow* this, GlobalContext* globalCtx) {
     }
 }
 
+bool EnCow_IsSecondCowInTower(EnCow* this, GlobalContext* globalCtx) {
+    return 
+        gSaveContext.n64ddFlag && 
+        globalCtx->sceneNum == SCENE_SOUKO && 
+        (this->actor.world.pos.x + this->actor.world.pos.z == -173 || this->actor.world.pos.x + this->actor.world.pos.z == -72);
+}
+
+bool EnCow_IsSecondCowInStable(EnCow* this, GlobalContext* globalCtx) {
+    return 
+        gSaveContext.n64ddFlag && 
+        globalCtx->sceneNum == SCENE_MALON_STABLE && 
+        (this->actor.world.pos.x + this->actor.world.pos.z == -257 || this->actor.world.pos.x + this->actor.world.pos.z == -138);
+}
+
+bool EnCow_IsSecondCowInGrotto(EnCow* this, GlobalContext* globalCtx) {
+    return gSaveContext.n64ddFlag && globalCtx->sceneNum == SCENE_KAKUSIANA && this->actor.world.pos.x + this->actor.world.pos.z == 3194;
+}
+
+bool EnCow_IsSecondCow(EnCow* this, GlobalContext* globalCtx) {
+    return EnCow_IsSecondCowInTower(this, globalCtx) || EnCow_IsSecondCowInStable(this, globalCtx) || EnCow_IsSecondCowInGrotto(this, globalCtx);
+}
+
+GetItemID EnCow_GetItemFromCow(EnCow* this, GlobalContext* globalCtx, bool setFlag) {
+    GetItemID itemId = ITEM_NONE;
+
+    if (Inventory_HasEmptyBottle()) {
+        return GI_MILK;
+    }
+
+    s32 collectableFlag = 0x1A;
+    if (EnCow_IsSecondCow(this, globalCtx)) {
+        collectableFlag = 0x19;
+    }
+
+    if (gSaveContext.n64ddFlag && !Flags_GetCollectible(globalCtx, collectableFlag)) {
+        itemId = Randomizer_GetRandomizedItemId(GI_MILK, this->actor.id, this->actor.world.pos.x + this->actor.world.pos.z, globalCtx->sceneNum);
```

Comment:
just curious, what happens if we just pass the actor params in here? are they the same for every cow in the scene?

Reply:
Looking at the changes in randomizer.cpp, it does indeed look like cows have their own params. Which makes sense because the game needs to know where to place them. Would also make it not break if you ever decide to move cows again in the future.

Reply:
Copying from other comment: actorParams for all cows is the same, from what I can tell each cow has two actors, one is the visual cow with actorParam 0 and the second is the collision box with actorParam 1.

Reply:
I wouldn't call this quite resolved yet. I've thought about it and x+z seems like it has pretty high potential to not be unique. As a simplified example, a cow at (10, 10) and a cow at (2, 18) will appear identical to that function. There's gotta be a better way to handle this, keep in mind that if we need to, adding properties to the cow actor is also an option.

Reply:
So I've still got to use the coords to differentiate two cows within the same scene, but now for the randomizer.cpp calls I'm passing in 4540 + cowId  as the actorParam (454 is the cow actor ID, so with 9 cows it will be 4540-4548).

Still not great, but better than passing in the X + Z as actorParam

Reply:
No longer calling Randomizer_GetRandomizedItemId, using RandomizerCheck now.